ale/drivers/isis_spice_driver.py

```diff
@@ -0,0 +1,588 @@
+from glob import glob
+import os
+import struct
+import re
+
+import pvl
+import numpy as np
+import quaternion
+
+from ale import config
+from ale.drivers.base import Driver, Isis3
+
+def read_table_data(table_label, file):
+    """
+    Helper function to read all of the binary table data
+
+    Parameters
+    ----------
+    table_label : PVLModule
+        The ISIS table label
+    file : file
+        The ISIS cube file
+
+    Returns
+    -------
+    bytes
+      The binary portion of the table data
+    """
+    file.seek(label['StartByte']-1) # This -1 is straight out of ISIS
+    return file.read(label['Bytes'])
+
+def field_size(field_label):
+    """
+    Helper function to determine the size of a binary
+    table field
+
+    Parameters
+    ----------
+    field_label : PVLModule
+        The field label
+
+    Returns
+    -------
+    int
+      The size of the one entry in bytes
+    """
+    data_sizes = {
+        'Integer' : 4,
+        'Double'  : 8,
+        'Real'    : 4,
+        'Text'    : 1
+    }
+    return data_sizes[field_label['Type']] * field_label['Size']
+
+def field_format(field_label):
+    """
+    Helper function to get the format string for a
+    single entry in a table field
+
+    Parameters
+    ----------
+    field_label : PVLModule
+        The field label
+
+    Returns
+    -------
+    str
+      The format string for the entry binary
+    """
+    data_formats = {
+        'Integer' : 'i',
+        'Double'  : 'd',
+        'Real'    : 'f'
+    }
+    return data_formats[field_label['Type']] * field_label['Size']
+
+def parse_field(field_label, data, encoding='latin_1'):
+    """
+    Parses all of the entries for a table field into a list.
+    If there is only one entry return that entry outside of
+    a list.
+
+    Parameters
+    ----------
+    field_label : PVLModule
+        The field label
+
+    data : bytes
+        The binary data for the entire table
+
+    Returns
+    -------
+    list
+      The table field entries
+    """
+    if field_label['Type'] == 'Text':
+        results[field_label['Name']].append(data[:field_label['Size']].decode(encoding=encoding))
+    else:
+        data_format = field_format(field_label)
+        field_data = struct.unpack_from(data_format, data)
+        if len(field_data) == 1:
+            field_data = field_data[0]
+    return field_data
+
+def parse_table_data(table_label, data):
+    """
+    Parses an ISIS table into a dict where the keys are the
+    field names and the values are lists of entries.
+
+    Parameters
+    ----------
+    table_label : PVLModule
+        The table label
+
+    data : bytes
+        The binary data for the entire table
+
+    Returns
+    -------
+    dict
+      The table as a dict
+    """
+    fields = table_label.getlist('Field')
+    results = {field['Name']:[] for field in fields}
```

Comment:
Are these keys guaranteed to be unique?

Reply:
Yes, Table field labels always contain one of each:
* Name
* Type
* Size

Reply:
Also, the Table is invalid if two fields share the same name